### ISSUE TITLE
Internals and VST plugins not appearing in patchbay with Carla 2.0.b6 (Windows 32 Bits No-SSE)

### ISSUE THREAD

**Issue by E-Gwen** [falkTX/Carla#582]
Hi,
 
When you want to use a VST or a native plugin with Carla, you have no way to plug it to another plugn in patchbay.
 
Did I miss something, or what ?
 
E-Gwen.
 
P.S. : It is the same issue with Carla 1.2.4 and Carla 2.0.b5.

**Comment by falkTX** [comment]
Not sure if I get this...
Are you using the Rack engine mode, thus only seeing a single "Carla" object in the patchbay?

**Comment by E-Gwen** [comment]
Well, I was using the"Continuous Rack", because it was the way for me of using Carla With Jack.
Carla was appearing through Jack using this way, and I was thinking it was the only right thing to do if I wanted to use Carla with Jack.

Using the "Single Client" or the "multiple-client", I can now see "Carla" and the plugins appearing when I add some.
And only after this operation, Carla, or the plugins themshelves, appear through Jack, so I can control plugs with it.

Therefore, can you tell me what are the advantages and purposes of using "Single Client", "Multiple-Clients", or "Continuous Rack", by giving an internet link, for example, so I clearly understand in which case you have to use them ?

Thank you very much for your support. I was lost.
There is no bug!

E-Gwen.

**Comment by falkTX** [comment]
rack mode makes it easy to manage connections, because there are not any to be made.
the plugins just get processed in order, top to bottom.
so you can have something like:
1. midi filter
2. synth
3. fx1
4. fx2

those 4 plugins run in order, without having to manage connections between them.

There is no documentation for carla yet, though I made a small presentation about it where I go through some of its features, here: https://media.ccc.de/v/sonoj2017-1704-carla-plugin-host

I will start documentation once the beta phase is finished and I start doing RCs.

**Comment by E-Gwen** [comment]
Thank you.
It's a little clearer.

By the way, I can now use my Boss FC-50 footswitch with ZynReverb plugin, but not with 3 band EQ plugin. Is that on purpose, or is it a general limitation ?

Do you know where I can grab an OSC file, .touchosc or .js file, so I can control Carla with TouchOSC, or with Open Stage Cotrol ?

Best regards.

**Comment by falkTX** [comment]
hmm I wrote a comment here before but it did not get sent...

there is no touchosc config file adapted for carla that I know of.

The issue with footswitch seems weird one, but please report a new issue for that.
I am guessing it sends program changes, for which zynreverb has a few, but 3bandeq has not.
or maybe I misunderstood. anyway, new bug for that one.

closing this bug as it seems to me the original issue was just missing documentation
